libtransmission/announcer.cc

```diff
@@ -1244,14 +1248,22 @@ void on_scrape_error(tr_session const* /*session*/, tr_tier* tier, char const* e
     // switch to the next tracker
     current_tracker = tier->useNextTracker();
 
-    // schedule a rescrape
-    auto const interval = current_tracker->getRetryInterval();
-    auto const* const host_cstr = current_tracker->host.c_str();
-    tr_logAddDebugTier(
-        tier,
-        fmt::format("Tracker '{}' scrape error: {} (Retrying in {} seconds)", host_cstr, errmsg, interval));
     tier->lastScrapeSucceeded = false;
-    tier->scheduleNextScrape(interval);
+
+    if (current_tracker != nullptr)
```

Comment:
Same remark here:
No need for a `if`, we should have an assertion instead:
`TR_ASSERT(current_tracker != nullptr)`

Reply:
replacing the `if` with `TR_ASSERT(current_tracker != nullptr)` doesn't prevent the warning when you compile with `-DCMAKE_BUILD_TYPE=RelWithDebInfo`:
```
/.../libtransmission/announcer.cc: In function ‘void {anonymous}::on_scrape_done_helpers::on_scrape_error(const tr_session*, tr_tier*, const char*)’:
/.../libtransmission/announcer.cc:357:20: warning: potential null pointer dereference [-Wnull-dereference]
  357 |             return nullptr;
      |                    ^~~~~~~

```

TR_ASSERT() is a nop macro in this case.

Reply:
Well, what about `assert(current_tracker != nullptr)`? Although, not sure what is the coding practice that is desired. @ckerr any suggestion?

Reply:
Courtesy pong, I did see this question & am not ignoring y'all.

I am really busy the next few days but will try to give this a look tomorrow night.

Reply:
`assert(current_tracker != nullptr)` doesn't get rid of the warning either, btw. It's stripped away just like `TR_ASSERT` when `NDEBUG` is defined.